sealer delete panic when the file does not exist

```
[root@test-baremetal008 ~]# sealer delete -f .sealer/my-cluster/Clusterfile
2021-06-04 11:43:41 [EROR] apply cloud cluster failed open .sealer/my-cluster/Clusterfile: no such file or directory
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1b071b7]

goroutine 1 [running]:
github.com/alibaba/sealer/sealer/cmd.glob..func3(0x3b70360, 0xc000144680, 0x0, 0x2)
	/home/runner/work/sealer/sealer/src/github.com/alibaba/sealer/sealer/cmd/delete.go:32 +0x47
github.com/spf13/cobra.(*Command).execute(0x3b70360, 0xc000144660, 0x2, 0x2, 0x3b70360, 0xc000144660)
	/home/runner/work/sealer/sealer/src/github.com/alibaba/sealer/vendor/github.com/spf13/cobra/command.go:854 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x3b71860, 0x444bca, 0x3aea8c0, 0xc000000180)
	/home/runner/work/sealer/sealer/src/github.com/alibaba/sealer/vendor/github.com/spf13/cobra/command.go:958 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/work/sealer/sealer/src/github.com/alibaba/sealer/vendor/github.com/spf13/cobra/command.go:895
github.com/alibaba/sealer/sealer/cmd.Execute()
	/home/runner/work/sealer/sealer/src/github.com/alibaba/sealer/sealer/cmd/root.go:46 +0x31
main.main()
	/home/runner/work/sealer/sealer/src/github.com/alibaba/sealer/sealer/main.go:20 +0x20
```

--- a/apply/apply.go
+++ b/apply/apply.go
@@ -18,7 +18,6 @@ import (
 	"fmt"
 
 	"github.com/alibaba/sealer/common"
-	"github.com/alibaba/sealer/logger"
 	v1 "github.com/alibaba/sealer/types/api/v1"
 	"github.com/alibaba/sealer/utils"
 )
@@ -28,14 +27,20 @@ type Interface interface {
 	Delete() error
 }
 
-func NewApplierFromFile(clusterfile string) Interface {
-	cluster := &v1.Cluster{}
-	if err := utils.UnmarshalYamlFile(clusterfile, cluster); err != nil {
-		logger.Error("apply cloud cluster failed", err)
-		return nil
+func NewApplierFromFile(clusterfile string) (Interface, error) {
+	cluster, err := GetClusterFromFile(clusterfile)
+	if err != nil {
+		return nil, err
 	}
-	cluster.SetAnnotations(common.ClusterfileName, clusterfile)
-	return NewApplier(cluster)
+	return NewApplier(cluster), nil
+}
+
+func GetClusterFromFile(filepath string) (cluster *v1.Cluster, err error) {
+	cluster = &v1.Cluster{}
+	if err = utils.UnmarshalYamlFile(filepath, cluster); err != nil {
+		return nil, fmt.Errorf("failed to get cluster from %s, %v", filepath, err)
+	}
+	return
 }
 
 func NewApplier(cluster *v1.Cluster) Interface {

--- a/apply/cloud_applier.go
+++ b/apply/cloud_applier.go
@@ -78,7 +78,7 @@ func (c *CloudApplier) Apply() error {
 	cluster := c.ClusterDesired
 	clusterCurrent, err := GetCurrentCluster()
 	if err != nil {
-		return fmt.Errorf("failed to get current clcuster %v", err)
+		return fmt.Errorf("failed to get current cluster %v", err)
 	}
 
 	cloudProvider, err := infra.NewDefaultProvider(cluster)

--- a/apply/default_applier.go
+++ b/apply/default_applier.go
@@ -225,10 +225,10 @@ func (c *DefaultApplier) diff() (todoList []ActionName, err error) {
 	c.NodesToJoin, c.NodesToDelete = utils.GetDiffHosts(c.ClusterCurrent.Spec.Nodes, c.ClusterDesired.Spec.Nodes)
 	todoList = append(todoList, MountImage)
 	todoList = append(todoList, MountRootfs)
-	if c.MastersToJoin != nil || c.MastersToDelete != nil {
+	if len(c.MastersToJoin) > 0 || len(c.MastersToDelete) > 0 {
 		todoList = append(todoList, ApplyMasters)
 	}
-	if c.NodesToJoin != nil || c.NodesToDelete != nil {
+	if len(c.NodesToJoin) > 0 || len(c.NodesToDelete) > 0 {
 		todoList = append(todoList, ApplyNodes)
 	}
 	todoList = append(todoList, CNI)

--- a/sealer/cmd/apply.go
+++ b/sealer/cmd/apply.go
@@ -30,7 +30,12 @@ var applyCmd = &cobra.Command{
 	Short:   "apply a kubernetes cluster",
 	Example: `sealer apply -f Clusterfile`,
 	Run: func(cmd *cobra.Command, args []string) {
-		if err := apply.NewApplierFromFile(clusterFile).Apply(); err != nil {
+		applier, err := apply.NewApplierFromFile(clusterFile)
+		if err != nil {
+			logger.Error(err)
+			os.Exit(1)
+		}
+		if err = applier.Apply(); err != nil {
 			logger.Error(err)
 			os.Exit(1)
 		}

--- a/sealer/cmd/delete.go
+++ b/sealer/cmd/delete.go
@@ -52,7 +52,12 @@ var deleteCmd = &cobra.Command{
 				}
 			}
 		}
-		if err := apply.NewApplierFromFile(clusterFile).Delete(); err != nil {
+		applier, err := apply.NewApplierFromFile(clusterFile)
+		if err != nil {
+			logger.Error(err)
+			os.Exit(1)
+		}
+		if err = applier.Delete(); err != nil {
 			logger.Error(err)
 			os.Exit(1)
 		}

--- a/seautil/cmd/apply.go
+++ b/seautil/cmd/apply.go
@@ -23,11 +23,7 @@ import (
 	"github.com/alibaba/sealer/logger"
 )
 
-type ApplyFlag struct {
-	ClusterFile string
-}
-
-var applyFlag *ApplyFlag
+var clusterFile string
 
 // applyCmd represents the apply command
 var applyCmd = &cobra.Command{
@@ -35,17 +31,19 @@ var applyCmd = &cobra.Command{
 	Short: "apply a kubernetes cluster",
 	Long:  `seautil apply -f cluster.yaml`,
 	Run: func(cmd *cobra.Command, args []string) {
-		applier := apply.NewApplierFromFile(applyFlag.ClusterFile)
-		err := applier.Apply()
+		applier, err := apply.NewApplierFromFile(clusterFile)
 		if err != nil {
 			logger.Error(err)
-			os.Exit(-1)
+			os.Exit(1)
+		}
+		if err = applier.Apply(); err != nil {
+			logger.Error(err)
+			os.Exit(1)
 		}
 	},
 }
 
 func init() {
-	applyFlag = &ApplyFlag{}
 	rootCmd.AddCommand(applyCmd)
-	applyCmd.Flags().StringVarP(&applyFlag.ClusterFile, "clusterfile", "f", "", "cluster file filepath")
+	applyCmd.Flags().StringVarP(&clusterFile, "clusterfile", "f", "", "cluster file filepath")
 }

